AnnoDesigner.Core/DataStructures/QuadTree.cs

```diff
@@ -357,154 +359,184 @@ public void ReIndex(Func<T, Rect> boundsSelector)
         public QuadTree(Rect extent)
         {
             root = new Quadrant(extent);
-            Extent = extent;
         }
 
         /// <summary>
         /// Insert a <typeparamref name="T"/> item into the QuadTree.
         /// </summary>
-        /// <param name="item"></param>
-        /// <param name="bounds"></param>
-        public void Insert(T item, Rect bounds)
+        public void Add(T item)
         {
-            if (!root.Extent.Contains(bounds))
-            {
-                throw new ArgumentException($"{nameof(bounds)} of {nameof(item)} is greater than the extent of the quadtree.");
-            }
-            root.Insert(item, bounds);
+            var bounds = item.Bounds;
+            EnsureBounds(bounds);
+
+            root.Add(item, bounds);
         }
 
         /// <summary>
         /// Remove a <typeparamref name="T"/> item from the QuadTree.
         /// </summary>
-        /// <param name="item"></param>
-        /// <param name="bounds"></param>
-        public void Remove(T item, Rect bounds)
+        public bool Remove(T item)
         {
-            var quadrant = root.GetContainingQuadrant(bounds);
-            quadrant.Remove((item, bounds));
+            return root.Remove(item);
         }
 
         /// <summary>
-        /// Retrieves a sequence of items that intersect the given bounds.
+        /// Removes all items from the QuadTree.
         /// </summary>
-        /// <param name="bounds">The bounds to check</param>
-        /// <returns></returns>
-        public IEnumerable<T> GetItemsIntersecting(Rect bounds)
+        public void Clear()
         {
-            var items = new List<T>(previousCount);
-            root.GetItemsIntersecting(items, bounds);
-            previousCount = items.Count;
-            return items;
+            foreach (var item in this.ToList())
+            {
+                Remove(item);
+            }
         }
 
+        public bool Contains(T item) => root.Contains(item);
+
+        public void CopyTo(T[] array, int arrayIndex) => root.CopyTo(array, arrayIndex);
```

Comment:
Same as above, this feels like a weird method to have.

Reply:
Same as above, needed for ICollection<T> interface